packages/lesswrong/server/rateLimitUtils.ts

```diff
@@ -55,11 +56,16 @@ function getPostRateLimitInfos(
   recentKarmaInfo: RecentKarmaInfo
 ): Array<RateLimitInfo> {
   // for each rate limit, get the next date that user could post  
-  const userPostRateLimitInfo = getUserRateLimitInfo(userPostRateLimit, postsInTimeframe)
+  const userPostRateLimitInfo = getManualRateLimitInfo(userPostRateLimit, postsInTimeframe)
+
+  const features = {
+    ...recentKarmaInfo, 
+    downvoteRatio: getDownvoteRatio(user)
+  } 
 
   const autoRatelimits = forumSelect(autoPostRateLimits)
   const autoRateLimitInfos = autoRatelimits?.map(
-    rateLimit => getAutoRateLimitInfo(user, rateLimit, postsInTimeframe, recentKarmaInfo)
+    rateLimit => getAutoRateLimitInfo(user, features, rateLimit, postsInTimeframe, recentKarmaInfo)
```

Comment:
if `recentKarmaInfo` is fully a subset of `features`, probably we don't need be passing both in, and `getAutoRateLimitInfo` can be refactored to only use `features`.

Reply:
man, I really don't like the name `features`.